src/libPMacc/include/math/complex/Complex.hpp

```diff
@@ -0,0 +1,205 @@
+/**
+ * 2013-2015 Heiko Burau, Rene Widera, Richard Pausch, Alexander Debus
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace PMacc
+{
+namespace math
+{
+
+/** A complex number class */
+template<typename Type>
+struct Complex
+{
+
+public:
+
+    typedef Type type;
+
+    // constructor (real, imaginary)
+    HDINLINE Complex(Type real, Type imaginary = 0.0) : real(real), imaginary(imaginary) { }
```

Comment:
Please use the typedef `type` or add a new `typedef T_Type Type` after the class template type is renamed.